### ISSUE TITLE
Free with invalid size in the function imlib_free_image_and_decache()

### ISSUE THREAD

**Issue by Hee-won** [derf/feh#711]
Hi there.

The latest version (3.10) of feh causes memory free of invalid size when executed with a crafted input via argument -l.

We suspect that there is no proper size check of the size to be freed or reallocated in the functions of the Imlib2 library called by gib_imlib_free_image_and_decache().

Here is the output of the gdb results.

### Bug Report

(gdb) r
Starting program: /home/ubuntu/targets/feh-3.10/src/feh -l /home/ubuntu/pfuzz/feh/crashes/0x01fa84cb_0xb1c1261c
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
free(): invalid next size (normal)

Program received signal SIGABRT, Aborted.
__GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
50	../sysdeps/unix/sysv/linux/raise.c: No such file or directory.
(gdb) bt
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
#1  0x00007ffff7a21859 in __GI_abort () at abort.c:79
#2  0x00007ffff7a8c26e in __libc_message (action=action@entry=do_abort, fmt=fmt@entry=0x7ffff7bb6298 "%s\n")
    at ../sysdeps/posix/libc_fatal.c:155
#3  0x00007ffff7a942fc in malloc_printerr (str=str@entry=0x7ffff7bb86b8 "free(): invalid next size (normal)") at malloc.c:5347
#4  0x00007ffff7a95e3c in _int_free (av=0x7ffff7bebb80 <main_arena>, p=0x5555555b0320, have_lock=<optimized out>) at malloc.c:4322
#5  0x00007ffff7c165d5 in __imlib_FreeData () from /lib/x86_64-linux-gnu/libImlib2.so.1
#6  0x00007ffff7c169a3 in ?? () from /lib/x86_64-linux-gnu/libImlib2.so.1
#7  0x00007ffff7c16d3d in ?? () from /lib/x86_64-linux-gnu/libImlib2.so.1
#8  0x00007ffff7c16dff in ?? () from /lib/x86_64-linux-gnu/libImlib2.so.1
#9  0x00007ffff7bfca16 in imlib_free_image_and_decache () from /lib/x86_64-linux-gnu/libImlib2.so.1
#10 0x000055555555f596 in gib_imlib_free_image_and_decache (im=<optimized out>) at gib_imlib.c:121
#11 0x000055555555e53d in feh_file_info_load (file=file@entry=0x5555555a3940, im=im@entry=0x0) at filelist.c:386
#12 0x000055555555e634 in feh_file_info_preload (list=0x555555587460) at filelist.c:316
#13 0x000055555555e89e in feh_prepare_filelist () at filelist.c:491
#14 0x000055555556efb9 in init_parse_options (argc=<optimized out>, argv=<optimized out>) at options.c:953
#15 0x000055555555c1d0 in main (argc=3, argv=0x7fffffffe538) at main.c:57

### Envionment
OS: Ubuntu 20.04.5 LTS / 22.04 LTS
Release: Feh 3.10
Library: imlib2 1.6.1 and imlib2 1.7.4
Program: feh

### How to reproduce
$ feh -l poc-file
poc-file is attached.
[poc-file.txt](https://github.com/derf/feh/files/11719765/poc-file.txt)

**Comment by N-R-K** [comment]
This one is also caused by the TGA loader and has been fixed in imlib2 v1.10.0.

Backtrace from ASan confirming the TGA loader to be the cause of the memory corruption:

```
AddressSanitizer:DEADLYSIGNAL
=================================================================
==14242==ERROR: AddressSanitizer: SEGV on unknown address 0x627000040c88 (pc 0x7fb39cd02270 bp 0x00000000005f sp 0x7ffc2e5641f8 T0)
==14242==The signal is caused by a READ memory access.
    #0 0x7fb39cd02270 in tgaflip src/modules/loaders/loader_tga.c:489
    #1 0x7fb39cd02d6e in load2 src/modules/loaders/loader_tga.c:456
    #2 0x7fb39e8ef7ac in __imlib_LoadImageWrapper src/lib/image.c:300
    #3 0x7fb39e8efd15 in __imlib_LoadImage src/lib/image.c:513
    #4 0x7fb39e8d5704 in imlib_load_image_frame src/lib/api.c:890
```

**Comment by derf** [comment]
Hi, thanks for the report and the follow-up clarification!

As the issue lies within Imlib2 (and not feh) and has already been fixed in the latest Imlib2 release, I'm closing this issue. Feel free to re-open it if there's anything I missed.
